v2/internal/controllers/edge_case_test.go

```diff
@@ -221,3 +176,97 @@ func Test_CreateStorageAccountWithoutRequiredProperties_Rejected(t *testing.T) {
 	// Create it again
 	tc.CreateResourcesAndWait(acctCopy)
 }
+
+func Test_AzureName_IsImmutableOnceSuccessfullyCreated(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	acct := createStorageAccount(tc, rg)
+	tc.CreateResourcesAndWait(acct)
+
+	// Patch the account to change AzureName
+	newAzureName := "test123"
+	old := acct.DeepCopy()
+	acct.Spec.AzureName = newAzureName
+	
+	err := tc.PatchAndExpectError(old, acct)
+
+	tc.Expect(err).ToNot(BeNil())
+	tc.Expect(old.Spec.AzureName).ToNot(BeIdenticalTo(newAzureName))
+
+	// Delete the account
+	tc.DeleteResourceAndWait(acct)
+
+}
+
+func Test_Owner_IsImmutableOnceSuccessfullyCreated(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	acct := createStorageAccount(tc, rg)
+	tc.CreateResourcesAndWait(acct)
+
+	rg2 := tc.CreateTestResourceGroupAndWait()
+	// Patch the account to change AzureName
+	old := acct.DeepCopy()
+	acct.Spec.Owner = testcommon.AsOwner(rg2)
+
+	err := tc.PatchAndExpectError(old, acct)
+
+	tc.Expect(err).ToNot(BeNil())
+	tc.Expect(old.Owner().Name).ToNot(BeIdenticalTo(rg2.Name))
+
+	// Delete the account
+	tc.DeleteResourceAndWait(acct)
+
+}
+
+func Test_Owner_IsNotImmutableIfSuccessfullyNotCreated(t *testing.T) {
```

Comment:
You should be able to test this for `AzureName` too by first attempting to use an `AzureName` that's invalid and then correct it? For example if you try to make a storage account named `-------` I think you'll get rejected. You can then patch it to have a different name which should work.